### PR TITLE
Updating dockerfiles for Windows

### PR DESCRIPTION
**Pull request by brycem**
Cleaned up RELEASE version Dockerfiles for Windows

### CODE REVIEW

**Comment by lzybkr** [comment]
Can you document how the image `brycem/win10build` was built?

We are discussing not building the C/C++ Windows components in PowerShell/PowerShell and instead pull packages from NuGet - would this eliminate the need for your custom image?

**Comment by brycem** [comment]
That would be a much cleaner solution. Heres the image and a link to the dockerfile in git https://hub.docker.com/r/brycem/win10build/

Outlook für Android herunterladen




On Mon, Jan 23, 2017 at 1:10 PM -0800, "Jason Shirk" <notifications@github.com<mailto:notifications@github.com>> wrote:


Can you document how the image brycem/win10build was built?

We are discussing not building the C/C++ Windows components in PowerShell/PowerShell and instead pull packages from NuGet - would this eliminate the need for your custom image?

**Comment by brycem** [comment]
@lzybkr - Where did you guys shake out on removing the Standalone Win10SDK dependency?  It would definitely simplify.  Looks like CMake is available in NuGet as well.

**Comment by mirichmo** [comment]
I filed issue #3014, but it hasn't progressed beyond that.

**Comment by mirichmo** [comment]
@raghushantha - Can you please take a look?

File: docker/release/nanoserver/Dockerfile [hunk 1]
```diff
@@ -1,48 +1,44 @@
...
-      Remove-Item -Path $ZipFile ; `
-      New-Item -Type SymbolicLink -Path $PsFolder\ -Name Core -Value $PsFolder `
-    } else { throw 'Installation failed!  See c:\Dockerfile.log' } ;
+ADD $POWERSHELL_ZIP /PowerShell-win10-win2k16-x64.zip
+RUN [System.IO.FileInfo]$ZipFile = Get-Item -Path ./powershell-win10-win2k16-x64.zip;`
+    [String]$ZipHash = (Get-FileHash  -Path $ZipFile -Algorithm SHA256 | select -ExpandProperty Hash);`
+    If ($ZipHash -notmatch $Env:POWERSHELL_SHA256){`
+        Throw [String]$('['+$ZipHash+'] does not match ['+$Env:POWERSHELL_SHA256+']!')`
+    }`
+    [System.IO.DirectoryInfo]$PsFolder=New-Item -Path $Env:ProgramFiles\PowerShell -ItemType Directory -Force;`
```
Comment (mirichmo) [inline docker/release/nanoserver/Dockerfile]: Expand-Archive would simplify this. [Here](https://github.com/PowerShell/PowerShell/blob/master/docs/installation/windows.md#online-deployment-of-powershell-core) is an example.

**Review by mirichmo — CHANGES_REQUESTED** [review]

File: docker/release/windowsservercore/Dockerfile [hunk 2]
```diff
@@ -1,48 +1,45 @@
...
 FROM microsoft/windowsservercore
-MAINTAINER brycem@microsoft.com
-LABEL Readme.md="https://github.com/PowerShell/PowerShell/blob/master/docker/README.md"
-LABEL Description="This Dockerfile will install the latest release of PS."
+LABEL Readme.md="https://github.com/PowerShell/PowerShell/blob/master/docker/README.md",`
+      Description="This Dockerfile will install the latest release of PS."
 
-ARG POWERSHELL_MSI=https://github.com/PowerShell/PowerShell/releases/download/v6.0.0-alpha.14/PowerShell_6.0.0.14-alpha.14-win10-x64.msi
-ARG POWERSHELL_SHA256=503F3AD52223699765895D3E9615FBD7988194693BCB725BE90C9EF0CD594447
+ARG POWERSHELL_MSI=https://github.com/PowerShell/PowerShell/releases/download/v6.0.0-alpha.15/PowerShell_6.0.0-alpha.15-win10-win2k16-x64.msi
```
Comment (raghushantha) [inline docker/release/windowsservercore/Dockerfile]: @brycem Please pick up the latest release MSI here
https://github.com/PowerShell/PowerShell/releases/latest

Then the experience would be similar to Linux. There we simply apt-get install / yum install from packages.microsoft.com and the image is automatlically populated with latest PS

This comment applies all locations where the MSI is downloaded

File: docker/release/windowsservercore/Dockerfile [hunk 3]
```diff
@@ -1,48 +1,45 @@
...
+ARG POWERSHELL_SHA256=CC52D21F3287E412B9C3B73C98BB5B06F8056D49D63201072216DF92B7F2E59B
```
Comment (raghushantha) [inline docker/release/windowsservercore/Dockerfile]: Not really required to validate the hash. The MSI is authenticode signed & you are downloading from an HTTPS location

**Review by raghushantha — CHANGES_REQUESTED** [review]

**Comment by brycem** [comment]
Looks like @TravisEz13 has updated the URLs - I'll close this PR

**Pull request closed without merging**